test_data/pil/simple_div.pil

```diff
@@ -0,0 +1,38 @@
+constant %N = 0x10000;
+
+namespace SimpleDiv(%N);
+    col fixed BYTE(i) { i & 0xff };
+    col fixed TWOBYTE(i) { i & 0xffff };
+
+    col witness X;
+    col witness Y;
+    col witness Z;
+    col witness R;
+
+    macro is_byte(V) { { V } in { BYTE }; };
+    macro declare_byte(V) { col witness V; is_byte(V); }
+
+    { X } in { TWOBYTE };
+    { Y } in { TWOBYTE };
+    { Z } in { TWOBYTE };
+    { R } in { TWOBYTE };
+
+    // Compute X = Y / Z, i.e.
+    // X * Z + R = Y, where 0 <= R < Z
+
+    Z = 3;
+    Y = 13;
+
+    X * Z + R = Y;
+    Z - R - 1 = Y_b1 + Y_b2 * 0x100,
+    X = X_b1 + X_b2 * 0x100,
```

Comment:
This inline implementation of division yields simpler identities than what we get if instead division is implemented inside an instruction, and the instruction is then used in the code.

As an example, running the test `instruction_tests::divu` in the current version of PR #320  (a19c5cef1d03a7822fd7245e582edf79c897c894), we get the following polynomial identity:
```
(Assembly.instr_divu * (Assembly.X - (((Assembly.X_b5 + (Assembly.X_b6 * 256)) + (Assembly.X_b7 * 65536)) + (Assembly.X_b8 * 16777216)))) = 0
```
Notice the whole thing is multiplied by the instruction selector `Assembly.instr_divu`, which won't happen in this example where the polynomial is defined inline.

Reply:
I think this is good enough since the selector will be a constant number at some point during witness generation.